### ISSUE TITLE
Keypoints are misaligned by up to half a pixel when an image dimension is not a multiple of 4

### ISSUE THREAD

**Issue by f-dy** [davnords/LoMa#15]
Same root cause as Parskatt/dad#18: LoMa carries its own copy of the DaD detector in `src/loma/detector/dad.py`, so it shows the same behaviour and can be fixed here independently of upstream.

`to_pixel_coords` maps a score-map index `i` to `i + 0.5` on a `[0, W]` axis, so a horizontal flip maps `x` to `W - x` exactly, and detections on an image and on its mirror must agree to the subpixel. They do not, unless the dimension is a multiple of 4: the error reaches 0.5 px, following `W % 4` on x and `H % 4` on y independently.

`DaD.load_image` floors both dimensions with `(scale * W) // 8 * 8`, so that path never hits this. Any caller passing its own tensor to `detect` does, including the proposed hloc extractor in cvg/Hierarchical-Localization#495, which forwards hloc's resized image at the source aspect ratio.

### Cause

`forward_impl` upsamples the logits back through the sizes recorded by the encoder. Those are floor-halvings, so a dimension that does not halve exactly makes an `F.interpolate` ratio non-integral, and with `align_corners=False` the score map lands off the pixel grid. Only the two finest levels matter in practice, which is why the condition is `% 4` and not `% 8`:

| width | %8 | %4 | upsample ratios (fine to coarse) | biased |
|---|---|---|---|---|
| 497 | 1 | 1 | **2.0040**, 2.0000, 2.0000 | yes |
| 498 | 2 | 2 | 2.0000, **2.0081**, 2.0000 | yes |
| 499 | 3 | 3 | **2.0040**, **2.0081**, 2.0000 | yes |
| 500 | 4 | 0 | 2.0000, 2.0000, **2.0161** | no |
| 504 | 0 | 0 | 2.0000, 2.0000, 2.0000 | no |

500 and 508 are the discriminating cases: not multiples of 8, but unbiased, because only the coarsest ratio is non-integral and that level is smoothed away.

### Reproduction

```python
"""Keypoints are not flip-equivariant unless the image dimensions are multiples of 4."""

import numpy as np
import torch
from scipy.spatial import cKDTree
from scipy.stats import trim_mean
from skimage import data

from loma.detector.dad import DaD

detector = DaD(DaD.Cfg(compile=False)).eval().cuda()


def pixel_keypoints(image):
    tensor = torch.from_numpy(image.transpose(2, 0, 1)[None] / 255.0).float().cuda()
    with torch.inference_mode():
        kpts = detector.detect({"image": tensor}, num_keypoints=2048)["keypoints"][0]
    coords = detector.to_pixel_coords(kpts, image.shape[0], image.shape[1])
    return coords.cpu().numpy().astype(np.float64)


base = np.asarray(data.astronaut())[..., :3]
print(" W    W%4   x offset (expected +0.000)")
for width in range(497, 513):
    crop = np.ascontiguousarray(base[:512, :width])
    original = pixel_keypoints(crop)
    mirrored = pixel_keypoints(np.ascontiguousarray(crop[:, ::-1]))
    mirrored[:, 0] = width - mirrored[:, 0]
    distance, index = cKDTree(original).query(mirrored, k=1)
    near = distance < 2.0
    offset = (mirrored[near] - original[index[near]])[:, 0]
    flag = "" if abs(trim_mean(offset, 0.2)) < 0.05 else "   <-- biased"
    print(f" {width}   {width % 4}     {trim_mean(offset, 0.2):+.3f}{flag}")
```

Output on a Tesla T4, torch 2.14.0+cu130, LoMa at 8fb59c4:

```
 W    W%4   x offset (expected +0.000)
 497   1     -0.366   <-- biased
 498   2     -0.087   <-- biased
 499   3     -0.481   <-- biased
 500   0     -0.002
 501   1     -0.388   <-- biased
 502   2     -0.103   <-- biased
 503   3     -0.495   <-- biased
 504   0     +0.004
 505   1     -0.369   <-- biased
 506   2     -0.088   <-- biased
 507   3     -0.487   <-- biased
 508   0     -0.001
 509   1     -0.386   <-- biased
 510   2     -0.103   <-- biased
 511   3     -0.488   <-- biased
 512   0     +0.003
```

Sweeping the height instead gives the same pattern on y (`-0.432`, `-0.117`, `-0.549` for `H % 4` of 1, 2, 3). The axes are independent: across 16 widths the y residual holds at `-0.008`, and across 16 heights the x residual holds at `+0.005`.

### Suggested fix

Padding the bottom-right up to a multiple of 4, then cropping the score map back, removes the bias at every size and leaves already-correct sizes bit-identical. Padding to 8 instead would also make the coarsest halving exact, but that was measured to move keypoints by only 0.0025 px, so 4 is the smallest sufficient modulus. Overhead is under 0.2% of a `detect` call (0.04 ms of ~23 ms on a T4). #16 does that, with a test covering one size per residue class on both axes.
